tests/integration/devfile/utils/utils.go

```diff
@@ -186,3 +190,132 @@ func ExecWithRestartAttribute(projectDirPath, cmpName, namespace string) {
+				} else {
+					helper.ReplaceString(filepath.Join(context, "server.js"), "Hello", "Hello odo")
+				}
+			}
+		}
+	}()
+
+	if !isDevfileTest {
+		flag = strings.TrimSpace(fmt.Sprintf("%s-app -v 4 %s", odoV1Watch.SrcType, flag))
+	}
+
+	success, err := helper.WatchNonRetCmdStdOut(
+		("odo watch " + flag + " --context " + context),
+		time.Duration(5)*time.Minute,
+		func(output string) bool {
+			if isDevfileTest {
+				stringsMatched := true
+
+				for _, stringToBeMatched := range odoV2Watch.StringsToBeMatched {
+					if !strings.Contains(output, stringToBeMatched) {
+						stringsMatched = false
+					}
+				}
+
+				if stringsMatched {
+					// Verify delete from component pod
+					getContainerExecListdir(odoV1Watch, odoV2Watch, runner, platform, project, isDevfileTest)
+					return true
+				}
+			} else {
+				curlURL := helper.CmdShouldPass("curl", odoV1Watch.RouteURL)
+				if strings.Contains(curlURL, "Hello odo") {
+					// Verify delete from component pod
+					getContainerExecListdir(odoV1Watch, odoV2Watch, runner, platform, project, isDevfileTest)
+				}
+				return strings.Contains(curlURL, "Hello odo")
+			}
+
+			return false
+		},
+		startSimulationCh,
+		func(output string) bool {
+			return strings.Contains(output, "Waiting for something to change")
+		})
+
+	Expect(success).To(Equal(true))
+	Expect(err).To(BeNil())
+
+	if !isDevfileTest {
+		// Verify memory limits to be same as configured
+		getMemoryLimit := runner.(helper.OcRunner).MaxMemory(odoV1Watch.SrcType+"-app", odoV1Watch.AppName, project)
+		Expect(getMemoryLimit).To(ContainSubstring("700Mi"))
+		getMemoryRequest := runner.(helper.OcRunner).MinMemory(odoV1Watch.SrcType+"-app", odoV1Watch.AppName, project)
+		Expect(getMemoryRequest).To(ContainSubstring("400Mi"))
+	}
+}
+
+func getContainerExecListdir(odoV1Watch OdoV1Watch, odoV2Watch OdoV2Watch, runner interface{}, platform, project string, isDevfileTest bool) string {
+	var stdOut string
+	sourcePath := "/projects/nodejs-web-app"
+
+	if platform == "kube" {
+		ocRunner := runner.(helper.OcRunner)
+		if isDevfileTest {
+			podName := ocRunner.GetRunningPodNameByComponent(odoV2Watch.CmpName, project)
+			stdOut = ocRunner.ExecListDir(podName, project, sourcePath)
+		} else {
+			podName := ocRunner.GetRunningPodNameOfComp(odoV1Watch.SrcType+"-app", project)
+			envs := ocRunner.GetEnvs(odoV1Watch.SrcType+"-app", odoV1Watch.AppName, project)
+			dir := envs["ODO_S2I_SRC_BIN_PATH"]
+			stdOut = ocRunner.ExecListDir(podName, project, filepath.Join(dir, "src"))
+		}
+	} else if platform == "docker" {
+		dockerRunner := runner.(helper.DockerRunner)
+		containers := dockerRunner.GetRunningContainersByCompAlias(odoV2Watch.CmpName, "runtime")
+		Expect(len(containers)).To(Equal(1))
+		stdOut = dockerRunner.ExecContainer(containers[0], "ls -la "+sourcePath)
+	}
+
+	Expect(stdOut).To(ContainSubstring(("a.txt")))
+	Expect(stdOut).To(ContainSubstring((".abc")))
+	Expect(stdOut).To(Not(ContainSubstring(("abcd"))))
```

Comment:
How  if we check for `stdout` in a single line like https://github.com/openshift/odo/blob/master/tests/helper/helper_generic.go#L73